File: backend/test_ws.js
const WebSocket = require('ws');
const API_TOKEN = '6945c91024ee47.12893127'; // User's actual API key from .env

const wsStocks = new WebSocket(`wss://ws.eodhistoricaldata.com/ws/us?api_token=${API_TOKEN}`);
const wsForex = new WebSocket(`wss://ws.eodhistoricaldata.com/ws/forex?api_token=${API_TOKEN}`);

console.log('🔍 4 Sembol Test Başlıyor...\n');

wsStocks.onopen = () => {
  console.log('✅ [STOCKS] Bağlandı (NASDAQ & DAX)');
  wsStocks.send(JSON.stringify({
    "action": "subscribe",
    "symbols": "NDX,^IXIC,^GDAXI,DAX,QQQ,EWG,USOIL,XTIUSD,CL.COMM,CL.F,BZ.F"
  }));
};

wsStocks.onmessage = (e) => {
  const data = JSON.parse(e.data);
  console.log('[STOCKS]', data.s, '| Fiyat:', data.p, '| Hacim:', data.v);
};

wsStocks.onerror = (e) => console.error('[STOCKS] Hata:', e);

wsForex.onopen = () => {
  console.log('✅ [FOREX] Bağlandı');
  wsForex.send(JSON.stringify({
    "action": "subscribe",
    "symbols": "XAUUSD,XTIUSD,USOIL.FOREX"
  }));
};

wsForex.onmessage = (e) => {
  const data = JSON.parse(e.data);
  console.log('[FOREX]', data.s, '| Fiyat:', data.p, '| Zaman:', new Date().toLocaleTimeString());
};

wsForex.onerror = (e) => console.error('[FOREX] Hata:', e);

setTimeout(() => {
  console.log('Test bitti, kapatılıyor...');
  wsStocks.close();
  wsForex.close();
  process.exit(0);
}, 10000); // 10 saniye bekle

File: backend/test_ws.py
import asyncio
import json
import logging
import websockets

API_TOKEN = '6989161fc498d3.47082906'

async def test_ws():
    print('🔍 4 Sembol Test Basliyor...\n')
    
    async def listen_stocks():
        try:
            async with websockets.connect(f"wss://ws.eodhistoricaldata.com/ws/us?api_token={API_TOKEN}") as ws:
                print('✅ [STOCKS] Baglandi (NASDAQ & DAX)')
                await ws.send(json.dumps({
                    "action": "subscribe",
                    "symbols": "NDX,^IXIC,^GDAXI,DAX,QQQ,EWG,USOIL,XTIUSD,CL.F,BZ.F"
                }))
                
                while True:
                    try:
                        message = await asyncio.wait_for(ws.recv(), timeout=5.0)
                        data = json.loads(message)
                        if data.get('s'):
                            print(f"[STOCKS] {data.get('s')} | Fiyat: {data.get('p')} | Hacim: {data.get('v')}")
                        else:
                            print(f"[STOCKS] {data}")
                    except asyncio.TimeoutError:
                        print("[STOCKS] Waiting...")
        except BaseException as e:
            print(f"[STOCKS] Hata: {e}")

    async def listen_forex():
        try:
            async with websockets.connect(f"wss://ws.eodhistoricaldata.com/ws/forex?api_token={API_TOKEN}") as ws:
                print('✅ [FOREX] Baglandi')
                await ws.send(json.dumps({
                    "action": "subscribe",
                    "symbols": "XAUUSD,XTIUSD,USOIL.FOREX"
                }))
                
                while True:
                    try:
                        message = await asyncio.wait_for(ws.recv(), timeout=5.0)
                        data = json.loads(message)
                        if data.get('s'):
                            print(f"[FOREX] {data.get('s')} | Fiyat: {data.get('p')}")
                        else:
                            print(f"[FOREX] {data}")
                    except asyncio.TimeoutError:
                        print("[FOREX] Waiting...")
        except BaseException as e:
            print(f"[FOREX] Hata: {e}")

    # Run for 15 seconds
    st_task = asyncio.create_task(listen_stocks())
    fx_task = asyncio.create_task(listen_forex())
    
    await asyncio.sleep(15)
    
    st_task.cancel()
    fx_task.cancel()
    print('Test bitti, kapatiliyor.')

if __name__ == '__main__':
    asyncio.run(test_ws())
